.changesets/feat_bnjjj_chore_4269.md

```diff
@@ -0,0 +1,45 @@
+### Promote subscription callback protocol to GA ([Issue #3884](https://github.com/apollographql/router/issues/3884))
+
+#### Configuration changes
+
+In order to promote subscription callback protocol to GA we finalized the specs and made some breaking changes. That's why when running this new version you'll have an error if you had `subscription.mode.preview_callback` in your configuration it won't start. You'll have to update your configuration and make sure your subgraph implementation is compliant with the stable version of subscription callback protocol. It's not longer `preview_callback` in the configuration but just `callback`. 
+
+Here is an example of configuration:
+
+```yaml
+subscription:
+  enabled: true
+  mode:
+    callback:
+      public_url: http://127.0.0.1:4000/custom_callback
+      listen: 0.0.0.0:4000
+      path: /custom_callback
+      heartbeat_interval: 5secs # can be "disabled", by default it's 5secs
+```
+
+One of the main difference in configuration is the behavior of `public_url`, this field must now include the full url of your callback endpoint, which means if you specified a path like in this example it's set to `/custom_callback` you'll have to also specify that path in the `public_url` field. Previously we automatically added the path to the `public_url` but not anymore. It will let you configure your own public url and use some redirection on your proxy if you have one in front of the router for example.
+We also added `heartbeat_interval` configuration field to configure specific heartbeat interval, the heartbeat can also be disabled by setting `disabled`.
```

Comment:
```suggestion
2. Update the `public_url` field to include the full URL of your callback endpoint.

    Previously in preview, the public URL used by the router was the automatic concatenation of the `public_url` and `path` fields. In GA, the behavior has changed, and the router uses exactly the value set in `public_url`. This enables you to configure your own public URL, for example if you have a proxy in front of the router and want to configure redirection with the public URL.
    
    In the example of the GA configuration below, the path `/custom_callback` is no longer automatically appended to `public_url`, so instead it has to be set explicitly as `public_url: http://127.0.0.1:4000/custom_callback`.

3. Configure the new `heartbeat_interval` field to set the period that a heartbeat must be sent to the callback endpoint for the subscription operation. 

    The default heartbeat interval is 5 seconds. Heartbeats can be disabled by setting `heartbeat_interval: disabled`.
    
```yaml
subscription:
  enabled: true
  mode:
    callback: #highlight-line
      public_url: http://127.0.0.1:4000/custom_callback
      listen: 0.0.0.0:4000
      path: /custom_callback
      heartbeat_interval: 5secs # can be "disabled", by default it's 5secs
```
```